File: 2025-09/2025-09-23/stringMirror.js
function isMirror(str1, str2) {

  return str1.replace(/[^a-zA-Z]/g, "") === 
    str2.replace(/[^a-zA-Z]/g, "")
        .split("")
        .reverse()
        .join("");
}

console.log(isMirror("helloworld", "helloworld"));
console.log(isMirror("Hello World", "dlroW olleH"));
console.log(isMirror("RaceCar", "raCecaR"));
console.log(isMirror("RaceCar", "RaceCar"));
console.log(isMirror("Mirror", "rorrim"));
console.log(isMirror("Hello World", "dlroW-olleH"));
console.log(isMirror("Hello World", "!dlroW !olleH"));

File: 2025-09/2025-09-23/stringMirror.py
def is_mirror(str1, str2):
    import re

    return re.sub(r"[^a-zA-Z]", "", str1) == re.sub(r"[^a-zA-Z]", "", str2)[::-1]

print(is_mirror("helloworld", "helloworld"))
print(is_mirror("Hello World", "dlroW olleH"))
print(is_mirror("RaceCar", "raCecaR"))
print(is_mirror("RaceCar", "RaceCar"))
print(is_mirror("Mirror", "rorrim"))
print(is_mirror("Hello World", "dlroW-olleH"))
print(is_mirror("Hello World", "!dlroW !olleH"))